src/functions.php

```diff
@@ -46,6 +47,26 @@
 use function MongoDB\BSON\toPHP;
 use function substr;
 
+/**
+ * Registers a PSR-3 logger to receive log messages from the driver/library.
+ *
+ * Adding a registered logger is a NOP.
+ */
+function add_logger(LoggerInterface $logger): void
```

Comment:
Seeing this made me realise that all functions use snake_case naming instead of camelCase. I'm not sure why we opted to do that, but I'd be in favour of changing that. Most functions in here are internal, with `with_transaction` and `is_document` being the lone exceptions.

I've created PHPLIB-1244 to track this, so feel free to leave the names as-is for consistency, but if we agree on this I'd like to take care of this before releasing 1.17.

Reply:
`with_transaction()` came much later, and probably should have been `withTransaction()` since we already had camelCase in PHPC at the time.

There's no good reason that other, internal functions were snake_case other than conforming to other PHP internal functions (`is_pipeline` vs. `is_int`).